plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java

```diff
@@ -195,4 +215,29 @@ public static Object[][] timestampTypes()
                 {"timestamp(12)"}
         };
     }
+
+    @DataProvider
+    public Object[][] testTableNameDataProvider()
+    {
+        return testTableNameTestData().stream()
+                .collect(toDataProvider());
+    }
+
+    private List<String> testTableNameTestData()
+    {
+        return ImmutableList.<String>builder()
+                .add("tricky")
+                .add("[tricky]")
+                .add("[tri[]cky]")
+                .add("[tri[ ]cky]")
+                .add("[tr; icky]")
+                .add("[tr i' 'cky]")
```

Comment:
also, why not immediately add all the test cases from 

```
        return ImmutableList.<String>builder()
                .add("lowercase")
                .add("UPPERCASE")
                .add("MixedCase")
                .add("an_underscore")
                .add("a-hyphen-minus") // ASCII '-' is HYPHEN-MINUS in Unicode
                .add("a space")
                .add("atrailingspace ")
                .add(" aleadingspace")
                .add("a.dot")
                .add("a,comma")
                .add("a:colon")
                .add("a;semicolon")
                .add("an@at")
                .add("a\"quote")
                .add("an'apostrophe")
                .add("a`backtick`")
                .add("a/slash`")
                .add("a\\backslash`")
                .add("adigit0")
                .add("0startwithdigit")
                .build();
```

?


are these cases superior over the list of cases from `io.trino.testing.BaseConnectorTest#testColumnNameTestData`?

having these two lists out of sync today will make test unification harder tomorrow, since someone will need to think about every each of these cases and think whether to add them to a merged result

it would be better to do this hard thinking part within this PR

- enhance cases in `io.trino.testing.BaseConnectorTest#testColumnNameTestData` method covering names with brackets
  - i don't think we need cases containing brackets AND apostrophes at the same time, but if you think these are valuable, let's add them too
- have a test here with exact same list of test cases to cover and with a TODO comment to move it to BCT.